godtools/App/Flows/ToolNavigation/ToolOpenedFrom.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  ToolOpenedFrom.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 4/24/25.
+//  Copyright © 2025 Cru. All rights reserved.
+//
+
+import Foundation
+
+enum ToolOpenedFrom {
+    case dashboardLessons
+    case dashboardFavoritesFeaturedLesson
+    case dashboardFavoritesFavoritedTool
+    case dashboardTools
+    case deepLink
+    case learnToShare
```

Comment:
Hey @levieggertcru, do you think these should be more granular?  Like distinguishing between opening from toolDetails,  whether learnToShare came from tools or favorites, etc.

Reply:
Hey @rachaelblue no let's not worry about getting too granular right now.  That will be enough to cover this related task.  

If we do need more granularity for any reason we can introduce that then.